rskj-core/src/main/java/co/rsk/net/eth/RskWireProtocol.java

```diff
@@ -151,29 +156,37 @@ public void channelRead0(final ChannelHandlerContext ctx, EthMessage msg) throws
      *************************/
 
     protected void processStatus(org.ethereum.net.eth.message.StatusMessage msg, ChannelHandlerContext ctx) throws InterruptedException {
-
         try {
-            Genesis genesis = GenesisLoader.loadGenesis(config, config.genesisInfo(), config.getBlockchainConfig().getCommonConstants().getInitialNonce(), true);
-            if (!Arrays.equals(msg.getGenesisHash(), genesis.getHash().getBytes())
-                    || msg.getProtocolVersion() != version.getCode()) {
+            if (msg.getProtocolVersion() != version.getCode()) {
                 loggerNet.info("Removing EthHandler for {} due to protocol incompatibility", ctx.channel().remoteAddress());
-                if (msg.getProtocolVersion() != version.getCode()){
-                    loggerNet.info("Protocol version {} - message protocol version {}", version.getCode(), msg.getProtocolVersion());
-                } else {
-                    loggerNet.info("Config genesis hash {} - message genesis hash {}", genesis.getHash(), msg.getGenesisHash());
-                }
+                loggerNet.info("Protocol version {} - message protocol version {}",
+                                version.getCode(), msg.getProtocolVersion());
                 ethState = EthState.STATUS_FAILED;
                 recordEvent(EventType.INCOMPATIBLE_PROTOCOL);
                 disconnect(ReasonCode.INCOMPATIBLE_PROTOCOL);
                 ctx.pipeline().remove(this); // Peer is not compatible for the 'eth' sub-protocol
                 return;
             }
 
-            if (config.networkId() != msg.getNetworkId()) {
-                loggerNet.info("Different network received: config network ID {} - message network ID {}", config.networkId(), msg.getNetworkId());
+            if (msg.getNetworkId() != config.networkId()) {
+                loggerNet.info("Removing EthHandler for {} due to invalid network", ctx.channel().remoteAddress());
+                loggerNet.info("Different network received: config network ID {} - message network ID {}",
+                               config.networkId(), msg.getNetworkId());
                 ethState = EthState.STATUS_FAILED;
                 recordEvent(EventType.INVALID_NETWORK);
                 disconnect(ReasonCode.NULL_IDENTITY);
+                ctx.pipeline().remove(this);
+                return;
+            }
+
+            if (!Arrays.equals(msg.getGenesisHash(), genesis.getHash().getBytes())) {
```

Comment:
Do you think you could make `getGenesisHash` return a `Keccak256` instance and do `genesis.getHash().equals(msg.getGenesisHash())`? You would also avoid `new`ing in the log below.

Reply:
I think i kind of fixed this.